src/text.rs

```diff
@@ -1,15 +1,80 @@
+use crate::util;
+use crate::Character;
+use num_traits::Bounded;
+
 /// A unique id identifying this text.
 #[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Hash)]
-pub struct TextId(u64);
+pub struct TextId(usize);
 
-impl From<u64> for TextId {
-    fn from(value: u64) -> Self {
+impl From<usize> for TextId {
+    fn from(value: usize) -> Self {
         TextId(value)
     }
 }
 
-impl From<TextId> for u64 {
-    fn from(value: TextId) -> u64 {
+impl From<TextId> for usize {
+    fn from(value: TextId) -> usize {
         value.0
     }
 }
+
+/// A text structure used as the target for pattern searching in the index.
+///
+/// Not only does it contain the text, but also the maximum character value in the
+/// text. This is used to determine the number of bits needed to store the
+/// characters in the text.
+pub struct Text<C, T>
```

Comment:
`Text` and `TextId` are not 1:1 now - the former represents a whole text to search, while the latter represents the ID of a _fragment_ of the text.

Maybe it's better to stop using the word "multi-texts" and coin other terms such as "multi-docs" and "DocId".